### PR TITLE
Minor test code cleanup

### PR DESCRIPTION
**Pull request by StefanSpieker**
Minor Test Code cleanup
* simplified asserts 
* use try with resources instead of try finally
* minor improvements on stream API calls
* removed some redundancies

### Proposed changelog entries

* N/A

### CODE REVIEW

File: test/src/test/java/hudson/scm/ChangeLogSetTest.java [hunk 1]
```diff
@@ -30,7 +31,7 @@ public void catchingExceptionDuringAnnotation() {
             notCaught = true;
         }
         assertEquals((new EntryImpl()).getMsg(), change.getMsg());
-        assertEquals(false, notCaught);
+        assertFalse(notCaught);
```
Comment (zbynek) [inline test/src/test/java/hudson/scm/ChangeLogSetTest.java]: Maybe kill the `notCaught` variable and use `fail(t.getMessage())` in the catch block instead.

File: test/src/test/java/hudson/model/ParametersAction2Test.java [hunk 2]
```diff
@@ -108,28 +108,28 @@ public void parametersDefinitionChange() throws Exception {
                 new StringParameterValue("undef", "undef")
         )));
 
-        assertTrue("undef parameter is not listed in getParameters",
-                !hasParameterWithName(build.getAction(ParametersAction.class), "undef"));
+        assertFalse("undef parameter is not listed in getParameters",
```
Comment (zbynek) [inline test/src/test/java/hudson/model/ParametersAction2Test.java]:
```suggestion
        assertFalse("undef parameter is not listed in getParameters",
```

File: test/src/test/java/hudson/model/ParametersAction2Test.java [hunk 3]
```diff
@@ -108,28 +108,28 @@ public void parametersDefinitionChange() throws Exception {
...
-        p.addProperty(new ParametersDefinitionProperty(Arrays.<ParameterDefinition>asList(
+        p.addProperty(new ParametersDefinitionProperty(Arrays.asList(
                 new StringParameterDefinition("foo", "foo"),
                 new StringParameterDefinition("bar", "bar"),
                 new StringParameterDefinition("undef", "undef"))));
 
         // undef is still not listed even after being added to the job parameters definition
-        assertTrue("undef parameter is not listed in getParameters",
-                !hasParameterWithName(build.getAction(ParametersAction.class), "undef"));
+        assertFalse("undef parameter is not listed in getParameters",
```
Comment (zbynek) [inline test/src/test/java/hudson/model/ParametersAction2Test.java]:
```suggestion
        assertFalse("undef parameter is not listed in getParameters",
```

File: test/src/test/java/hudson/scm/ChangeLogSetTest.java (hunk 1 above)
Comment (StefanSpieker) [reply test/src/test/java/hudson/scm/ChangeLogSetTest.java]: You are right - that is much clearer.

File: test/src/test/java/hudson/model/ParametersAction2Test.java (hunk 2 above)
Comment (StefanSpieker) [reply test/src/test/java/hudson/model/ParametersAction2Test.java]: Done 👍

File: test/src/test/java/hudson/model/ParametersAction2Test.java (hunk 3 above)
Comment (StefanSpieker) [reply test/src/test/java/hudson/model/ParametersAction2Test.java]: 👍

**Review by res0nance — APPROVED** [review]

File: test/src/test/java/hudson/model/ParametersAction2Test.java [hunk 4]
```diff
@@ -261,10 +257,10 @@ public void nonParameterizedJob() throws Exception {
                 new StringParameterValue("bar", "bar")
         )));
 
-        assertTrue("foo parameter is not listed in getParameters",
-                !hasParameterWithName(build.getAction(ParametersAction.class), "foo"));
-        assertTrue("bar parameter is not listed in getParameters",
-                !hasParameterWithName(build.getAction(ParametersAction.class), "bar"));
+        assertFalse("foo parameter is not listed in getParameters",
```
Comment (zbynek) [inline test/src/test/java/hudson/model/ParametersAction2Test.java]:
```suggestion
        assertFalse("foo parameter is not listed in getParameters",
```

File: test/src/test/java/hudson/model/ParametersAction2Test.java [hunk 5]
```diff
@@ -261,10 +257,10 @@ public void nonParameterizedJob() throws Exception {
                 new StringParameterValue("bar", "bar")
         )));
 
-        assertTrue("foo parameter is not listed in getParameters",
-                !hasParameterWithName(build.getAction(ParametersAction.class), "foo"));
-        assertTrue("bar parameter is not listed in getParameters",
-                !hasParameterWithName(build.getAction(ParametersAction.class), "bar"));
+        assertFalse("foo parameter is not listed in getParameters", 
+                hasParameterWithName(build.getAction(ParametersAction.class), "foo"));
+        assertFalse("bar parameter is not listed in getParameters",
```
Comment (zbynek) [inline test/src/test/java/hudson/model/ParametersAction2Test.java]:
```suggestion
        assertFalse("bar parameter is not listed in getParameters",
```

**Review by zbynek — APPROVED** [review]
two more trailing spaces, otherwise looks good.

**Review by oleg-nenashev — APPROVED** [review]
Thanks! 
I plan to merge it tomorrow if no negative feedback

**Pull request merged**